File: 3280/Solution_3280.java
import java.text.SimpleDateFormat;
import java.util.Date;

public class Solution {

    private String binary(int x) {
        StringBuilder s = new StringBuilder();
        for (; x != 0; x >>= 1) {
            s.append(x & 1);
        }
        return s.reverse().toString();
    }

    public String convertDateToBinary(String date) {
        int year = Integer.parseInt(date.substring(0, 4));
        int month = Integer.parseInt(date.substring(5, 7));
        int day = Integer.parseInt(date.substring(8, 10));
        return binary(year) + "-" + binary(month) + "-" + binary(day);
    }

    public static void main(String[] args) {
        // 获取当前日期
        Date today = new Date();

        // 将日期转换为字符串格式 YYYY-MM-DD
        SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd");
        String formattedDate = formatter.format(today);

        // 调用 convertDateToBinary 方法
        Solution solution = new Solution();
        System.out.println(solution.convertDateToBinary(formattedDate));
    }
}

File: 3280/Solution_3280.cpp
#include <iostream>
#include <string>
#include <algorithm>

using namespace std;

class Solution {
public:
    string binary(int x) {
        string s;
        while (x) {
            s.push_back('0' + (x & 1)); // 将二进制的最低位添加到字符串
            x >>= 1; // 右移一位
        }
        reverse(s.begin(), s.end()); // 反转字符串以获得正确的二进制顺序
        return s.empty() ? "0" : s;  // 如果x为0，返回"0"
    }

    string convertDateToBinary(string date) {
        int year = stoi(date.substr(0, 4));  // 提取年份
        int month = stoi(date.substr(5, 2)); // 提取月份
        int day = stoi(date.substr(8, 2));   // 提取日期
        return binary(year) + "-" + binary(month) + "-" + binary(day); // 拼接结果
    }
};

int main() {
    Solution solution;
    string date = "2025-01-02";
    cout << solution.convertDateToBinary(date) << endl; // 输出二进制日期
    return 0;
}

